Lib/hashlib.py

```diff
@@ -231,6 +231,8 @@ def file_digest(fileobj, digest, /, *, _bufsize=2**18):
     view = memoryview(buf)
     while True:
         size = fileobj.readinto(buf)
+        if size is None:
+            raise BlockingIOError("I/O operation would block.")
```

Comment:
We should use `IOError` instead of `BlockingIOError`

> Raised when an operation would block on an object (e.g. socket) set for non-blocking operation. Corresponds to errno [EAGAIN](https://docs.python.org/3/library/errno.html#errno.EAGAIN), [EALREADY](https://docs.python.org/3/library/errno.html#errno.EALREADY), [EWOULDBLOCK](https://docs.python.org/3/library/errno.html#errno.EWOULDBLOCK) and [EINPROGRESS](https://docs.python.org/3/library/errno.html#errno.EINPROGRESS).

Reply:
IOError is the same as OSError since Python 3.3, and I don't think we use those aliases now. For me, BlockingIOError is correct in the sense that `None` is being returned by `readinto` so it's as if we were simply re-raising the exception (for buffered IO, they may raise BlockingIOError; for raw IO, they may return None to indicate the same thing).